common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -845,33 +847,32 @@ private static int javaVersion0() {
         if (isAndroid0()) {
             majorVersion = 6;
         } else {
-            majorVersion = majorVersionFromJavaSpecificationVersion();
+            majorVersion = majorVersionFromSystemClass();
         }
 
         logger.debug("Java version: {}", majorVersion);
 
         return majorVersion;
     }
 
-    // Package-private for testing only
-    static int majorVersionFromJavaSpecificationVersion() {
-        return majorVersion(SystemPropertyUtil.get("java.specification.version", "1.6"));
-    }
-
-    // Package-private for testing only
-    static int majorVersion(final String javaSpecVersion) {
-        final String[] components = javaSpecVersion.split("\\.");
-        final int[] version = new int[components.length];
-        for (int i = 0; i < components.length; i++) {
-            version[i] = Integer.parseInt(components[i]);
-        }
-
-        if (version[0] == 1) {
-            assert version[1] >= 6;
-            return version[1];
-        } else {
-            return version[0];
+    private static int majorVersionFromSystemClass() {
+        InputStream in = ClassLoader.getSystemResourceAsStream("java/lang/ClassLoader.class");
```

Comment:
@xxDark can you link to the spec where all of this is explained ?

Reply:
> 
> 
> @xxDark can you link to the spec where all of this is explained ?

https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html

Reply:
> 
> 
> > @xxDark can you link to the spec where all of this is explained ?
> 
> https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html

So basically, we are skipping magic of class, mirror version, and then read major version

Reply:
> 
> 
> > > @xxDark can you link to the spec where all of this is explained ?
> > 
> > 
> > https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html
> 
> So basically, we are skipping magic of class, mirror version, and then read major version

Also a bit explanation why we substract 44:
Major version represented as 45, 46, 47, 48, and so on
so e.g:
For JDK 8 major version is 52; 52 - 44 = 8

Reply:
why not fallback to old method if getSystemResourceAsStream returns null?

Reply:
When will it return null? I don't think there is any possibility for that, but ok, will make fallback method

Reply:
Did you actually run into a situation where the old implementation fails?